### DISCUSSION TITLE
Cannot preview .json files

### DISCUSSION THREAD

**Discussion by EarthMessenger** [spencerwooo/onedrive-vercel-index#404] [Q&A] [answered]
无法预览json文件。报错如下：  
Unable to preview json file. The error is as follows:

![图片](https://user-images.githubusercontent.com/49364506/153405371-f20a91a2-d684-476f-abbf-a05d036cd014.png)

所fork的仓库：[EarthMessenger/onedrive-vercel-index](https://github.com/EarthMessenger/onedrive-vercel-index)  
The forked repository: [EarthMessenger/onedrive-vercel-index](https://github.com/EarthMessenger/onedrive-vercel-index)

样例 example: 
- [onedrive.earthmessenger.xyz/zh-CN/code-backup/.vscode/settings.json](//onedrive.earthmessenger.xyz/zh-CN/code-backup/.vscode/settings.json)
- [onedrive.earthmessenger.xyz/zh-CN/oi-workspace/.vscode/settings.json](//onedrive.earthmessenger.xyz/zh-CN/oi-workspace/.vscode/settings.json)
- [onedrive.earthmessenger.xyz/zh-CN/code-backup/.vscode/c_cpp_properties.json](//onedrive.earthmessenger.xyz/zh-CN/code-backup/.vscode/c_cpp_properties.json)
- and all the .json files in my deployment ([onedrive.earthmessenger.xyz](//onedrive.earthmessenger.xyz))

浏览器 browser：Firefox Developer Edition 98.0b2

**Comment by spencerwooo** [comment, marked as answer]
Partially fixed by f4a6eb53bc6f476f3abc4361aaf3819247beee7d

Highlighting is not working yet, bear with me. Trying to figure out a way to dynamically import all Prism supported languages.

**Reply by EarthMessenger** [reply]
thanks!

**Reply by spencerwooo** [reply]
Should be all working as expected now, check latest commit.

**Reply by EarthMessenger** [reply]
That’s nice. But I found that the file [https://onedrive.earthmessenger.xyz/zh-CN/oi-workspace/libreoj/%234.%20Quine/problem.json/problem.json](https://onedrive.earthmessenger.xyz/zh-CN/oi-workspace/libreoj/%234.%20Quine/problem.json/problem.json) still doesn’t preview correctly and returns 404. It can display correctly in its parent [folder](https://onedrive.earthmessenger.xyz/zh-CN/oi-workspace/libreoj/%234.%20Quine). Is that because of my OneDrive configuration?

**Reply by spencerwooo** [reply]
I believe so, same file is rendered correctly on my end. Check demo - https://drive.spencerwoo.com/%F0%9F%A5%9F%20Some%20test%20files/Code/problem.json

**Reply by EarthMessenger** [reply]
Perhaps it’s because of the character “#” in the folder name?

**Reply by myl7** [reply]
> That’s nice. But I found that the file https://onedrive.earthmessenger.xyz/zh-CN/oi-workspace/libreoj/%234.%20Quine/problem.json/problem.json still doesn’t preview correctly and returns 404. It can display correctly in its parent [folder](https://onedrive.earthmessenger.xyz/zh-CN/oi-workspace/libreoj/%234.%20Quine). Is that because of my OneDrive configuration?

Since the console of the page indicates HTTP status code 429, maybe you exceeded Vercel rate limit?
